### PR TITLE
Holding a tactical item (cardboard cutout, plants) now gives TRAIT_UNKNOWN, replaces your name with the name of the tactical item

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

I added TRAIT_UNKNOWN to the code for picking up and dropping potted plants. Picking them up will make you totally anonymous (and suspicious). This is the same thing that infiltrator suit and heretic shadow cloak does.

I also added in some code that changes your displayed name to the name of the item, and then I had to jiggle ID code a bit so it wouldn't display your worn ID; it'll now display the forced name of the tactical object.

## Why It's Good For The Game

Becoming an absolute master of disguise by hiding with a potted plant is hilarious. Causing people to become immensely paranoid of anyone holding a potted plant is very on-brand. Dropping the potted plant to reveal you're in full infiltrator gear or heretic aura is also a funny double bluff. The total anonymity afforded by this is balanced out by the fact that you have to use both hands to hold the plant so you can't defend yourself at all if you're making use of it.
## Changelog
:cl: Bisar
add: The interns remembered to water the plants around the station; their foilage is so thick that it totally hides the identity of anyone holding them!
/:cl:

### CODE REVIEW

**Comment by Inari-Whitebear** [comment]
Was this inspired by the ninja whose Stealth was borked and thus used potted plants?

**Comment by SomeoneYouProbablyKnow** [comment]
Shame you can't stand so still it removes you from sec/med hud altogether

**Comment by MrMelbert** [comment]
i don't like this because i remember seeing a detective ask the hop for an id named "potted plant" so they could disguise better

**Comment by Metekillot** [comment]
> i don't like this because i remember seeing a detective ask the hop for an id named "potted plant" so they could disguise better

HoP power creeping on the chameleon projector tbh

**Comment by Metekillot** [comment]
> Was this inspired by the ninja whose Stealth was borked and thus used potted plants?

That sounds hilarious

**Comment by SomeoneYouProbablyKnow** [comment]
> i don't like this because i remember seeing a detective ask the hop for an id named "potted plant" so they could disguise better

God fucking bless. Sad you can't do that against secuirty

**Comment by carlarctg** [comment]
I'm in favor of this because sec is hard to evade consistently when they're highly coordinated/bloodthirsty. It adds a great option.

**Comment by SomeoneYouProbablyKnow** [comment]
"Hard to evade sec" when i turn a corner, swap ids and color my clothes with a spraycan. D: Gta ahh sec
But yeah hiding in plants is funny and hiding in lockers is already a thing.

**Comment by Loafin34** [comment]
i'm sorry but it's way better for the potted plant to keep your ID it's just super funny that way

**Comment by SomeoneYouProbablyKnow** [comment]
It only effects sec anyway. No one's using the examine because of runechat and general game speed except the like 5 insane people. Wish it completely obscured you from the hud tbut that'd be a pretty big deal i feel like

**Comment by Ghommie** [comment]
I personally think this should be done by the tactical component, not the potted plant itself.

**Comment by Dawnseer** [comment]
Imagine not having a cardboard ID named potted plant

**Comment by Metekillot** [comment]
> I personally think this should be done by the tactical component, not the potted plant itself.

Good idea; I can put handling there to have the visible name be replaced by the name of the item, too.

> Shame you can't stand so still it removes you from sec/med hud altogether 

I've been putting off a total refactor of how clunky Agent cards are because TGUI scares me; I think I'll add consideration for this

**Comment by LSommers** [comment]
It would be better if it only did TRAIT_UNKNOWN if you don't have an ID, or something, that way shenanigans with cardboard IDs still work. It will never not be funny to have someone holding a plant and named VERY NORMAL PLANT sneaking around, y'know?

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 1]
```diff
@@ -110,6 +110,11 @@
 	var/obj/item/card/id/id = wear_id
 	if(HAS_TRAIT(src, TRAIT_UNKNOWN))
 		. = if_no_id //You get NOTHING, no id name, good day sir
+		var/list/identity = list(null, null, null)
+		SEND_SIGNAL(src, COMSIG_HUMAN_GET_VISIBLE_NAME, identity)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/human_helpers.dm:114]: Using the same signal in two places is usually no no. Make a new one for this case.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: config/maps.txt [hunk 2]
```diff
@@ -61,4 +61,5 @@ map multiz_debug
 endmap
 
 map runtimestation
+default
```
[the remark this answers is not in the document]
Comment (Metekillot) [reply config/maps.txt:64]: oop~

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 3]
```diff
@@ -103,6 +103,8 @@
 
 /obj/item/clothing/suit/armor/abductor/vest/proc/return_disguise_name(mob/living/carbon/human/source, list/identity)
 	SIGNAL_HANDLER
+	if(identity[VISIBLE_NAME_FORCED]) // name-forcing overrides disguise
+		return
```
Comment (MrMelbert) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:107]: This is entirely at the whim of the order the signals are registered, which stinks and is a bad pattern

Comment (Metekillot) [reply code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:107]: I addressed this particular array in all the places I found that it's manipulated. Did I miss some or is there a flaw in my logic? I did several rounds of testing.

**Pull request merged**